rerun_py/src/arrow.rs

```diff
@@ -148,31 +133,33 @@ pub fn build_chunk_from_components(
         .collect();
 
     // Extract the component data
-    let (arrays, fields): (Vec<ArrowArrayRef>, Vec<ArrowField>) = itertools::process_results(
-        components_per_descr.iter().map(|(component_descr, array)| {
-            array_to_rust(&array, &descriptor_to_rust(&component_descr)?)
-        }),
-        |iter| iter.unzip(),
-    )?;
+    let (arrays, component_descrs): (Vec<ArrowArrayRef>, Vec<ComponentDescriptor>) =
+        itertools::process_results(
+            components_per_descr.iter().map(|(component_descr, array)| {
+                let component_descr = descriptor_to_rust(&component_descr)?;
+                array_to_rust(&array).map(|array| (array, component_descr))
+            }),
+            |iter| iter.unzip(),
+        )?;
 
     let components: Result<Vec<(ComponentDescriptor, _)>, ChunkError> = arrays
         .into_iter()
-        .zip(fields)
-        .map(|(value, field)| {
-            let batch = if let Some(batch) = value.downcast_array_ref::<ArrowListArray>() {
+        .zip(component_descrs)
+        .map(|(list_array, descr)| {
+            let batch = if let Some(batch) = list_array.downcast_array_ref::<ArrowListArray>() {
                 batch.clone()
             } else {
                 let offsets =
-                    ArrowOffsetBuffer::from_lengths(std::iter::repeat(1).take(value.len()));
-                let field = ArrowField::new("item", value.data_type().clone(), true).into();
-                ArrowListArray::try_new(field, offsets, value, None).map_err(|err| {
+                    ArrowOffsetBuffer::from_lengths(std::iter::repeat(1).take(list_array.len()));
+                let field = ArrowField::new("item", list_array.data_type().clone(), true).into();
+                ArrowListArray::try_new(field, offsets, list_array, None).map_err(|err| {
                     ChunkError::Malformed {
                         reason: format!("Failed to wrap in List array: {err}"),
                     }
                 })?
             };
 
-            Ok((ComponentDescriptor::new(field.name().clone()), batch))
+            Ok((descr, batch))
```

Comment:
That line really was the heart of the issue. Everything else is just simplifications from there.